src/metric-engine/src/engine/open.rs

```diff
@@ -94,6 +94,21 @@ impl MetricEngineInner {
         Ok(responses)
     }
 
+    // If the metadata region is opened with a stale manifest,
+    // the metric engine may fail to recover logical tables from the metadata region,
+    // as the manifest could reference files that have already been deleted
+    // due to compaction operations performed by the region leader.
```

Comment:
I'm not sure we can repair the metadata in this case; it appears to be expected and may occur.